platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -344,6 +345,10 @@ public DefaultConfiguration(
 
         // Until the role at creation is set, changing usage won't trigger warnings
         this.roleAtCreation = roleAtCreation;
+
+        this.extendsFrom = domainObjectCollectionFactory.newDomainObjectSet(Configuration.class);
+        this.extendsFrom.whenObjectAdded(this::addConfigurationToExtendsFrom);
```

Comment:
❌ This listener will most certainly eagerly resolve any provider passed to `extendsFrom(Provider)`. It would be better to resolve the full `DomainObjectSet` when the `Configuration` is locked.

Reply:
Yes, this was intentional to not change the configuration implementation and to only change the user API. Full support of lazy configuration of the Configuration would be another (complex) PR. Or do you require this change here too?